src/pybind/mgr/cephadm/services/cephadmservice.py

```diff
@@ -286,6 +290,18 @@ def get_config_and_keyring(self,
             'keyring': keyring,
         }
 
+    def remove_keyring(self, daemon: DaemonDescription):
+        daemon_id: str = daemon.daemon_id
+        host: str = daemon.hostname
+
+        entity = self.get_auth_entity(daemon_id, host=host)
+
+        logger.info(f'Remove keyring: {entity}')
+        ret, out, err = self.mgr.check_mon_command({
+            'prefix': 'auth rm',
```

Comment:
what happens, if the keyring is missing. is `auth rm` going to fail or silently ignore?

Reply:
correct, `auth rm` appears to be idempotent